sources/api/static-pods/src/static_pods.rs

```diff
@@ -54,13 +55,27 @@ where
     S2: AsRef<[u8]>,
 {
     let name = name.as_ref();
-
     let dir = Path::new(STATIC_POD_DIR);
     fs::create_dir_all(&dir).context(error::Mkdir { dir: &dir })?;
-    let path = dir.join(name);
-    // Create the file if it does not exist, completely replace its contents
-    // if it does (constitutes an update).
-    fs::write(path, manifest).context(error::ManifestWrite { name })?;
+
+    // Create the pod manifest file as a temporary file first and finish writing to it
+    // before swapping any files out
+    let mut temp_manifest_file =
+        NamedTempFile::new_in(STATIC_POD_DIR).context(error::CreateTempfile)?;
```

Comment:
```suggestion
        NamedTempFile::new_in(&dir).context(error::CreateTempfile)?;
```
I think it doesn't have to recreate the Path this way.  Also makes it clearer to the reader that it's not a different place.